app/controllers/stock_takings_controller.rb

```diff
@@ -10,46 +10,18 @@ def new
     StockArticle.undeleted.each { |a| @stock_taking.stock_changes.build(:stock_article => a) }
   end
 
+  def new_on_stock_article_create # See publish/subscribe design pattern in /doc.
+    stock_article = StockArticle.find(params[:stock_article_id])
+    @stock_change = StockChange.new(:stock_article => stock_article)
+
+    render :layout => false
```

Comment:
This line is just for consistency. It seems to work without this line. I have no idea if we should include that line or not.

Reply:
I think it's a good idea to include it, to make sure no encompassing layout is used, just the template itself.